[Issue] [MIOpen] Build failure: miopen_gtest fails due to undeclared `ck` types in `implicitgemm_ck_util.hpp`

## Summary
The CI/CD build fails for `miopen_gtest` due to `implicitgemm_ck_util.hpp` referencing `ck::half_t`, `ck::bhalf_t`, `ck::tf32_t` when `ck` is not declared. This is likely caused by missing CK include headers or the lack of guarding for CK support in the source file.

## Full log:
- Linux: https://github.com/ROCm/TheRock/actions/runs/24944514058/job/73057316766?pr=4828
- Windows: https://github.com/ROCm/TheRock/actions/runs/24944514058/job/73071290118?pr=4828  ( [log](https://therock-ci-artifacts.s3.amazonaws.com/24930772372-windows/logs/math-libs/gfx110X-all/MIOpen_build.log) )

## Context:
- **Platform:** `Linux`, `Windows`
- **Impacted Arcch:** `gfx110xall` , `gfx94x`


## Error Snippet
```
.../implicitgemm_ck_util.hpp:245:49: error: use of undeclared identifier 'ck'
  return FillValidKernelsIDs<DeviceOpPtrs<ck::half_t, ck::half_t>, CKArgs>(problem);
                                                ^
.../implicitgemm_ck_util.hpp:251:57: error: use of undeclared identifier 'ck'
  FillValidKernelsIDs<DeviceOpPtrs<float, ck::tf32_t>, CKArgs>(problem);
                                                        ^
.../implicitgemm_ck_util.hpp:258:49: error: use of undeclared identifier 'ck'
  return FillValidKernelsIDs<DeviceOpPtrs<ck::bhalf_t, ck::bhalf_t>, CKArgs>(problem);
                                                ^
```

## Regression Range
Failures began after the following changes in rocm-libraries:
https://github.com/ROCm/rocm-libraries/compare/f9a8d1c2f88812124f735f31a3b4a916cd601bc2...7cb164373be108531afe9d3a43dd012acb4adea2


## Likely Cause
- `implicitgemm_ck_util.hpp` references types in the `ck::` namespace without ensuring the appropriate CK header (`ck/utility/data_type.hpp`, `ck/ck.hpp`, or equivalent) is included.
- It's possible CK support is conditionally compiled but this header is not correctly guarded, or the gtest target is missing CK include paths.

## Impact:
Blocker for Rocm-libraries promotion: https://github.com/ROCm/TheRock/pull/4823

--- a/projects/miopen/src/ck_impl/implicitgemm_ck_util.hpp
+++ b/projects/miopen/src/ck_impl/implicitgemm_ck_util.hpp
@@ -239,6 +239,7 @@ template <template <typename, typename> class DeviceOpPtrs,
           typename ProblemDescriptionType = miopen::conv::ProblemDescription>
 std::vector<std::string> FillValidKernelsGeneric(const ProblemDescriptionType& problem)
 {
+#if MIOPEN_BACKEND_HIP && MIOPEN_USE_COMPOSABLEKERNEL
     switch(problem.GetInDataType())
     {
     case miopenHalf:
@@ -261,6 +262,10 @@ std::vector<std::string> FillValidKernelsGeneric(const ProblemDescriptionType& p
 
     default: return {};
     }
+#else
+    (void)problem;
+    return {};
+#endif
 }
 
 /**
@@ -344,6 +349,7 @@ template <template <typename, typename> class BilinearPtrs,
           typename ProblemDescriptionType = miopen::conv::ProblemDescription>
 std::vector<std::string> FillValidKernelsWithAlphaBetaGeneric(const ProblemDescriptionType& problem)
 {
+#if MIOPEN_BACKEND_HIP && MIOPEN_USE_COMPOSABLEKERNEL
     // Data type dispatch with TF32 support (C++17 compatible - uses helper function)
     switch(problem.GetInDataType())
     {
@@ -384,6 +390,10 @@ std::vector<std::string> FillValidKernelsWithAlphaBetaGeneric(const ProblemDescr
 
     default: return {};
     }
+#else
+    (void)problem;
+    return {};
+#endif
 }
 
 #if MIOPEN_BACKEND_HIP && MIOPEN_USE_COMPOSABLEKERNEL

